wordpress-mobile/WordPress-iOS
# Save for later: basic offline support

Fixes #14337

Review only after #14328 is merged

This PR adds basic support to offline when saving a post for later in the Reader.

Why *basic* support? What we do is:

1. We open a hidden webview to preload the post, so when offline must of the resources (mainly images) will be there
2. Why not all of them? Because the WebView respects the cache from the server. If an image comes from a server that disables caching its resources, this will not work (but it will work with WP.com)

### To test

**Make sure that you have overridden the new reader Feature Flag and it's not using the older Reader (Me -> App Settings -> Debug -> "Reader content displayed in a webview" should be ON)**

1. Open the Reader
2. Save a few posts for later
3. Close the app
4. Go offline
5. Open the app
6. Open the saved posts, most of the images should be there

You can use this snippet to make sure the webview content is erased:

```swift
let websiteDataTypes = NSSet(array: [WKWebsiteDataTypeDiskCache, WKWebsiteDataTypeMemoryCache])
let date = Date(timeIntervalSince1970: 0)
WKWebsiteDataStore.default().removeData(ofTypes: websiteDataTypes as! Set<String>, modifiedSince: date, completionHandler:{ })
```

PR submission checklist:
